### PR TITLE
Make keystore-password, keystore, trustore keys in secret configurable

### LINKED ISSUE #718: K8SSAND-1828 ⁃ Make keystore-password, keystore, trustore keys in secret configurable
Hi, i want to use **client encryption** in cassandra, and implement it via cert-manager+k8ssandra-operator. I create certificate via cert-manager and enable following options

```
spec:
  secretName: ssl-test-jks-keystore
  keystores:
    jks:
      create: true
      passwordSecretRef:
        key: keystore-pass
        name: jks-password
```

Then cert-manager add in secret `ssl-test-jks-keystore` with key `keystore.jks/truststore.jks`, i use this secret in k8ssandraCluster spec, to define my keystore/trustore via `clientEncryptionStores` section. But k8ssandra-operator and cert-manager hard-coded key name in secret. K8ssandra-operator expect - `keystore,trustore`, but cert-manager used `keystore.jks,trustore.jks` . 

Could you change default constants according default key of cert-manager or make key name configurable?

[https://github.com/cert-manager/cert-manager/blob/2db62c21c337cb5af3ea9dce1cbc0b69cfc7c509/pkg/controller/certificates/issuing/internal/keystore.go#L46
](https://github.com/cert-manager/cert-manager/blob/2db62c21c337cb5af3ea9dce1cbc0b69cfc7c509/pkg/controller/certificates/issuing/internal/keystore.go#L46
)
[https://github.com/k8ssandra/k8ssandra-operator/blob/32c4fcfdebc5b28b9c731bca8ada9ab79e3910a8/pkg/encryption/encryption_stores.go#L30
](https://github.com/k8ssandra/k8ssandra-operator/blob/32c4fcfdebc5b28b9c731bca8ada9ab79e3910a8/pkg/encryption/encryption_stores.go#L30)

Second problem, that `keystore-password` also hardcoded and expected in the same secret as `keystore`, cert-manager store it different secret, password secret managed by user, keystore secret managed by cert-managed. According above code keystore will be in secreet `ssl-test-jks-keystore` secret, password will be in `jks-password` secret

Could you extend of API `encyption.Store` to fix following cases and change defaults according cert-manager(or make native support of cert-manager via Certificate resources instead of Secret)?

┆Issue is synchronized with this [Jira Task](https://k8ssandra.atlassian.net/browse/K8SSAND-1828) by [Unito](https://www.unito.io)
┆friendlyId: K8SSAND-1828
┆priority: Medium

### PR DESCRIPTION
**Pull request by studypyth**
**What this PR does**:
Make keystore-password, keystore, trustore keys in secret configurable

**Which issue(s) this PR fixes**:
Fixes #718 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [x] Documentation added/updated
- [x] CHANGELOG.md updated (not required for documentation PRs)
- [x] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: pkg/encryption/encryption_stores.go [hunk 1]
```diff
@@ -6,15 +6,25 @@ import (
 
 // +kubebuilder:object:generate=true
 type Stores struct {
-	// ref to the secret that contains the keystore and its password
-	// the expected format of the secret is a "keystore" entry and a "keystore-password" entry
+	// ref to the secret that contains the keystore and its can contain password if not used keystorePasswordSecretRef
+	// if not specified key explicitly "keystore" entry and a "keystore-password" entry will be used
```
Comment (adejanovski) [inline pkg/encryption/encryption_stores.go:10]:
```suggestion
	// ref to the secret that contains the keystore and optionally its password (which can also be specified through 
the keystorePasswordSecretRef field)
	// if keys are not specified, "keystore" entry and a "keystore-password" entry will be used
```

File: pkg/encryption/encryption_stores.go [hunk 2]
```diff
@@ -6,15 +6,25 @@ import (
...
 	// +kubebuilder:validation:Required
-	KeystoreSecretRef corev1.LocalObjectReference `json:"keystoreSecretRef"`
+	KeystoreSecretRef *SecretKeySelector `json:"keystoreSecretRef"`
 
-	// ref to the secret that contains the truststore and its password
-	// the expected format of the secret is a "truststore" entry and a "truststore-password" entry
+	// ref to the secret that contains the truststore and can its contain password if not used truststoreSecretRef
+	// if not specified key explicitly "keystore" entry and a "keystore-password" entry will be used
```
Comment (adejanovski) [inline pkg/encryption/encryption_stores.go:15]:
```suggestion
	// ref to the secret that contains the truststore and optionally its password (which can also be specified through the truststorePasswordSecretRef field)
	// if keys are not specified explicitly, "keystore" entry and a "keystore-password" entry will be used
```

**Review by adejanovski — CHANGES_REQUESTED** [review]
Hi @studypyth,

thanks a lot for pushing this PR!
I love the fact that you managed to retain backwards compatibility while allowing to use separate secrets and keys for the stores and their passwords.
I have a small request for you about variable name changes, and if you wouldn't mind it would be great to update the docs [here](https://github.com/k8ssandra/k8ssandra-operator/blob/main/docs/encryption/README.md) to explain the new capabilities. 

Thanks again

**Comment by studypyth** [comment]
Hi @adejanovski, thanks for the review, I've updated the docs, fixed naming issues and updated CRD and autogenerated files from controller-gen(zz_generated.deepcopy.go)

File: pkg/cassandra/encryption.go [hunk 3]
```diff
@@ -204,13 +203,39 @@ func ReadEncryptionStoresSecrets(ctx context.Context, klusterKey types.Namespace
...
-	secretKey := types.NamespacedName{Namespace: namespace, Name: secretName}
-	if err := remoteClient.Get(ctx, secretKey, clientKeystoreSecret); err != nil {
+func ReadEncryptionStorePassword(ctx context.Context, namespace string, remoteClient client.Client, stores *encryption.Stores, storeName encryption.StoreName) (string, error) {
+	var general, specific *encryption.SecretKeySelector
+	var secretName, secretKey string
+
+	switch storeName {
+	case encryption.StoreNameKeystore:
+		general = stores.KeystoreSecretRef
+		specific = stores.KeystorePasswordRef
```
[the remark this answers is not in the document]
Comment (studypyth) [reply pkg/cassandra/encryption.go]: good point, i have renamed variables

File: pkg/encryption/encryption_stores.go (hunk 1 above)
Comment (studypyth) [reply pkg/encryption/encryption_stores.go]: Fixed

File: pkg/encryption/encryption_stores.go (hunk 2 above)
Comment (studypyth) [reply pkg/encryption/encryption_stores.go]: Fixed

**Pull request merged**